internal/subscriber/core.go

```diff
@@ -51,6 +54,8 @@ func (c *Core) NewSubscriber(ctx context.Context,
 	}
 
 	subsCtx, cancelFunc := context.WithCancel(ctx)
+	errChan := make(chan error, DefaultLimitForErrChannel)
```

Comment:
Why do we want to use a buffered channel here? Once an error is encountered we restart the subscriber anyway.

Reply:
existing code was creating a buffered channel, here I have just passed it to delay consumer.
But yeah, since we are reading from it at the same time in stream, we can use a unbuffered channel